llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -7834,6 +7843,61 @@ void SIInstrInfo::splitScalar64BitBFE(SIInstrWorklist &Worklist,
   addUsersToMoveToVALUWorklist(ResultReg, MRI, Worklist);
 }
 
+void SIInstrInfo::splitScalar64BitCountOp(SIInstrWorklist &Worklist,
+                                          MachineInstr &Inst, unsigned Opcode,
+                                          MachineDominatorTree *MDT) const {
+  //  (S_FLBIT_I32_B64 hi:lo) ->
+  // -> (umin (V_FFBH_U32_e32 hi), (uaddsat (V_FFBH_U32_e32 lo), 32))
```

Comment:
This looks better than the generic expansion for ctlz/cttz, maybe something similar could also be implemented there?

Also, RegBankSelect should get similar treatment for GlobalISel